Client/Frontend/Reader/ReaderMode.swift

```diff
@@ -128,6 +117,36 @@ struct ReadabilityResult {
             return nil
         }
     }
+
+    /// Encode to a dictionary, which can then for example be json encoded
+    func encode() -> [String:AnyObject] {
+        return ["domain": domain, "url": url, "content": content, "title": title, "credits": credits]
+    }
+
+    /// Initialize from a JSON encoded string
+    init?(string: String) {
+        let object = JSON(string: string)
+        let domain = object["domain"].asString
+        let url = object["url"].asString
+        let content = object["content"].asString
+        let title = object["title"].asString
+        let credits = object["credits"].asString
+
+        if domain == nil || url == nil || content == nil || title == nil || credits == nil {
+            return nil
+        }
+
+        self.domain = domain!
+        self.url = url!
+        self.content = content!
+        self.title = title!
+        self.credits = credits!
+    }
+
+    /// Encode to a JSON encoded string
+    func encode() -> String {
+        return JSON(encode() as [String:AnyObject]).toString(pretty: false)
```

Comment:
Hm, I'm surprised this is valid. We have two `encode()` functions with the same name and argument list returning different things? How does the compiler determine which version should be called? Maybe worth renaming to `encodeToString` and `encodeToDictionary` just to make things more explicit.